twine/commands/upload.py

```diff
@@ -48,6 +50,42 @@ def skip_upload(response, skip_existing, package):
             (response.status_code == 403 and msg_403 in response.text)))
 
 
+def check_status_code(response, verbose):
+    """
+    Additional safety net to catch response code 410 in case the
+    UploadToDeprecatedPyPIDetected exception breaks.
+    Also includes a check for response code 405 and prints helpful error
+    message guiding users to the right repository endpoints.
+    """
+    if (response.status_code == 410 and
+            response.url.startswith(("https://pypi.python.org",
```

Comment:
Another thought: this clause could be simpler and more flexible via `"pypi.python.org" in response.url`.

Reply:
Done in cf837a9.